java-checks/src/main/java/org/sonar/java/checks/regex/DuplicatesInCharacterClassCheck.java

```diff
@@ -19,59 +19,17 @@
  */
 package org.sonar.java.checks.regex;
 
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Set;
-import java.util.stream.Collectors;
 import org.sonar.check.Rule;
-import org.sonar.java.checks.helpers.SimplifiedRegexCharacterClass;
 import org.sonar.plugins.java.api.tree.ExpressionTree;
 import org.sonarsource.analyzer.commons.regex.RegexParseResult;
-import org.sonarsource.analyzer.commons.regex.ast.CharacterClassElementTree;
-import org.sonarsource.analyzer.commons.regex.ast.CharacterClassUnionTree;
-import org.sonarsource.analyzer.commons.regex.ast.RegexBaseVisitor;
-import org.sonarsource.analyzer.commons.regex.ast.RegexSyntaxElement;
+import org.sonarsource.analyzer.commons.regex.finders.DuplicatesInCharacterClassFinder;
 
 @Rule(key = "S5869")
 public class DuplicatesInCharacterClassCheck extends AbstractRegexCheck {
 
-  private static final String MESSAGE = "Remove duplicates in this character class.";
-
   @Override
   public void checkRegex(RegexParseResult regexForLiterals, ExpressionTree methodInvocationOrAnnotation) {
-    new DuplicateFinder().visit(regexForLiterals);
-  }
-
-  private class DuplicateFinder extends RegexBaseVisitor {
-
-    @Override
-    public void visitCharacterClassUnion(CharacterClassUnionTree tree) {
-      Set<RegexSyntaxElement> duplicates = new LinkedHashSet<>();
-      SimplifiedRegexCharacterClass characterClass = new SimplifiedRegexCharacterClass();
-      for (CharacterClassElementTree element : tree.getCharacterClasses()) {
-        List<RegexSyntaxElement> intersections = new SimplifiedRegexCharacterClass(element).findIntersections(characterClass);
-
-        if (!intersections.isEmpty()) {
-          for (RegexSyntaxElement intersection : intersections) {
-            if (!duplicates.contains(intersection)) {
-              // The element the current element is intersecting with should be included as well.
-              duplicates.add(intersection);
-            }
-          }
-          duplicates.add(element);
-        }
-        characterClass.add(element);
```

Comment:
The [code in Analyzer Commons](https://github.com/SonarSource/sonar-analyzer-commons/blob/f3adac98a3c8532a66cb64a4f1619d7a0a36fcd3/regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/DuplicatesInCharacterClassFinder.java#L58) is slightly different but is equivalent and simplier. (Note: we probably had this rather complex code because we were initially using a List instead of a Set).